demo_nodes_cpp/src/logging/use_logger_service.cpp

```diff
@@ -0,0 +1,220 @@
+// Copyright 2023 Sony Group Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <chrono>
+#include <memory>
+
+#include "rclcpp/rclcpp.hpp"
+
+#include "std_msgs/msg/string.hpp"
+
+#include "rcl_interfaces/srv/get_logger_levels.hpp"
+#include "rcl_interfaces/srv/set_logger_levels.hpp"
+
+using namespace std::chrono_literals;
+
+// This demo program shows how to enable logger service and control logger level via logger service.
+// Class LoggerServiceNode enable logger service and create a subscription. The callback of
+// subscription output received message by different log functions.
+// Class TestNode can set/get logger level of LoggerServiceNode and send message to it.
+
+class LoggerServiceNode : public rclcpp::Node
+{
+public:
+  explicit LoggerServiceNode(const std::string & node_name)
```

Comment:
`#include <string>` at the top of this file for this.

Reply:
Fixed it